torch/_inductor/compile_fx.py

```diff
@@ -268,6 +268,8 @@ def fake_tensor_prop(
             if not force_allow_non_fake_inputs
             else mock.patch.object(fake_mode, "allow_non_fake_inputs", True)
         )
+        # breakpoint()
+        fake_mode = torch._subclasses.FakeTensorMode(allow_non_fake_inputs=True)
```

Comment:
Obviously, this is not the right fix, but it's a workaround for the the problem I am running into. 

My setup is torch.export() with CPU example input, and inductor compile with CUDA sample input. 

When inductor runs fake tensor propagation at 
```
            FakeTensorProp(gm, mode=fake_mode).propagate_dont_convert_inputs(
                *example_inputs
            )
```
It complians
`RuntimeError: Unhandled FakeTensor Device Propagation for aten.mm.default, found two different devices cpu, cuda:0`
,even if all the `example_inputs` and module parameters and buffers are moved to the same device, 'cuda' in this case. 

As I traced through the code, I noticed that some cpu FakeTensor sneak into the program in the middle of `FakeTensorProp`, and I suspect this is coming from FakeTensorMode, originated from the torch.export() in CPU. 
So I tried this workaround, instead of using the`fake_mode` from example_input, I create a new one, and this worked. 

But I am still no sure what should be right fix? 

cc @ezyang and @jansel

Reply:
The faketensor used in FakeTensorProp is constructed from graph's node.meta['val'], which has the device and fake_mode used during torch.export(). 

I wonder if this is the proper way to construct fake_tensor.

Reply:
does this need to share the shape env with the existing fake env?

Reply:
In the internal production path, export and compilation are happening at different place and we don't serialized shape env. So I don't expect export and compilation to share the shape env.